pkg/cmd/pr/list/http.go

```diff
@@ -0,0 +1,247 @@
+			$owner: String!,
+			$repo: String!,
+			$limit: Int!,
+			$endCursor: String,
+			$baseBranch: String,
+			$labels: [String!],
+			$state: [PullRequestState!] = OPEN
+		) {
+			repository(owner: $owner, name: $repo) {
+				pullRequests(
+					states: $state,
+					baseRefName: $baseBranch,
+					labels: $labels,
+					first: $limit,
+					after: $endCursor,
+					orderBy: {field: CREATED_AT, direction: DESC}
+				) {
+					totalCount
+					nodes {
+						...pr
+					}
+					pageInfo {
+						hasNextPage
+						endCursor
+					}
+				}
+			}
+		}`
+
+	pageLimit := min(limit, 100)
+	variables := map[string]interface{}{
+		"owner":  repo.RepoOwner(),
+		"repo":   repo.RepoName(),
+		"labels": filters.Labels,
+	}
+
+	switch filters.State {
+	case "open":
+		variables["state"] = []string{"OPEN"}
+	case "closed":
+		variables["state"] = []string{"CLOSED", "MERGED"}
+	case "merged":
+		variables["state"] = []string{"MERGED"}
+	case "all":
+		variables["state"] = []string{"OPEN", "CLOSED", "MERGED"}
+	default:
+		return nil, fmt.Errorf("invalid state: %s", filters.State)
+	}
+
+	if filters.BaseBranch != "" {
+		variables["baseBranch"] = filters.BaseBranch
+	}
+
+	res := api.PullRequestAndTotalCount{}
+	var check = make(map[int]struct{})
+	client := api.NewClientFromHTTP(httpClient)
+
+loop:
+	for {
+		variables["limit"] = pageLimit
+		var data response
+		err := client.GraphQL(repo.RepoHost(), query, variables, &data)
+		if err != nil {
+			return nil, err
+		}
+		prData := data.Repository.PullRequests
+		res.TotalCount = prData.TotalCount
+
+		for _, pr := range prData.Nodes {
+			if _, exists := check[pr.Number]; exists {
+				continue
+			}
+			check[pr.Number] = struct{}{}
+
+			res.PullRequests = append(res.PullRequests, pr)
+			if len(res.PullRequests) == limit {
+				break loop
+			}
+		}
+
+		if prData.PageInfo.HasNextPage {
+			variables["endCursor"] = prData.PageInfo.EndCursor
+			pageLimit = min(pageLimit, limit-len(res.PullRequests))
+		} else {
+			break
+		}
+	}
+
+	return &res, nil
+}
+
+func searchPullRequests(httpClient *http.Client, repo ghrepo.Interface, filters prShared.FilterOptions, limit int) (*api.PullRequestAndTotalCount, error) {
+	if len(filters.Labels) > 1 {
+		return nil, fmt.Errorf("multiple labels with --assignee are not supported")
+	}
+
+	type response struct {
+		Search struct {
+			Nodes    []api.PullRequest
+			PageInfo struct {
+				HasNextPage bool
+				EndCursor   string
+			}
+			IssueCount int
+		}
+	}
+
+	query := fragment + `
+		query PullRequestSearch(
+			$q: String!,
+			$limit: Int!,
+			$endCursor: String,
+		) {
+			search(query: $q, type: ISSUE, first: $limit, after: $endCursor) {
+				issueCount
+				nodes {
+					...pr
+				}
+				pageInfo {
+					hasNextPage
+					endCursor
+				}
+			}
+		}`
+
+	q := githubsearch.NewQuery()
+	q.SetType(githubsearch.PullRequest)
+	q.InRepository(ghrepo.FullName(repo))
+	q.AddQuery(filters.Search)
+	q.SortBy(githubsearch.CreatedAt, githubsearch.Desc)
```

Comment:
Will this override any sort parameter the user might have specified in `filters.Search`?

Reply:
That's a great question. Any user's query will always be appended at the end, so their own `sort:*` directive would override this implicit `sort:` directive after ElasticSearch parses it. However, after some thought, I decided that since `sort:created-desc` is already the default for issue search, it doesn't really need to be explicitly specified here. I'll remove it 👍